viewing_party/party.py

```diff
@@ -1,23 +1,140 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if bool(title) and bool(genre) and bool(rating):
+        return {
+                    "title": title,
+                    "genre": genre,
+                    "rating":rating
+               }
+    else:
+        None
+
+def add_to_watched(user_data, movie):
+    
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+   
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    
+    for m_to_watch in user_data["watchlist"]:

```

Comment:
Note, while this may work here, it's considered dangerous practice to modify the content of a data structure that you're looping over.  Especially in more complex code, this can open you up to complex errors. A simple alternative here could be to save a copy of what you get from movie in user_data["watchlist"] and iterate over that, so that as you modify user_data["watchlist"], there's no risk.